From 18eed0f6f4e83eb94e0f8615ff8c73c23ced4412 Mon Sep 17 00:00:00 2001
From: Xiao Chen <xiao@apache.org>
Date: Tue, 15 Nov 2016 16:26:27 -0800
Subject: [PATCH 2511/2848] HADOOP-13815.
 TestKMS#testDelegationTokensOpsSimple and
 TestKMS#testDelegationTokensOpsKerberized Fails
 in Trunk.

(cherry picked from commit 61c0bedf775e6e794d4704485ec2c41a95aecae9)
(cherry picked from commit efc9ffc3b62a3de7660ab6fadeb59c8c2bca368b)
(cherry picked from commit cf366a0fd60d1b2db113cf4e252b64efd913239e)

Change-Id: Icb406d3d1c5c7a0e1f4f6fa5f5e942977aacd6b0
---
 .../hadoop/crypto/key/kms/server/TestKMS.java      |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/TestKMS.java b/hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/TestKMS.java
index 81e47b1..f8d5d6e 100644
--- a/hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/TestKMS.java
+++ b/hadoop-common-project/hadoop-kms/src/test/java/org/apache/hadoop/crypto/key/kms/server/TestKMS.java
@@ -1856,8 +1856,11 @@ public Void run() throws Exception {
                 Assert.fail("client should not be allowed to renew token with"
                     + "renewer=client1");
               } catch (Exception e) {
+                final DelegationTokenIdentifier identifier =
+                    (DelegationTokenIdentifier) token.decodeIdentifier();
                 GenericTestUtils.assertExceptionContains(
-                    "tries to renew a token with renewer", e);
+                    "tries to renew a token (" + identifier
+                        + ") with non-matching renewer", e);
               }
             }
 
-- 
1.7.9.5

